src/main/java/com/cupid/jikting/common/login/filter/CustomJsonUsernamePasswordAuthenticationFilter.java

```diff
@@ -0,0 +1,53 @@
+package com.cupid.jikting.common.login.filter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.security.authentication.AuthenticationServiceException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.util.StreamUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+public class CustomJsonUsernamePasswordAuthenticationFilter extends AbstractAuthenticationProcessingFilter {
+
+    private static final String DEFAULT_LOGIN_REQUEST_URL = "/members/login";
+    private static final String HTTP_METHOD = "POST";
+    private static final String CONTENT_TYPE = "application/json";
+    private static final String USERNAME_KEY = "username";
+    private static final String PASSWORD_KEY = "password";
+    private static final AntPathRequestMatcher DEFAULT_LOGIN_PATH_REQUEST_MATCHER =
+            new AntPathRequestMatcher(DEFAULT_LOGIN_REQUEST_URL, HTTP_METHOD);
+
+    private final ObjectMapper objectMapper;
+
+    public CustomJsonUsernamePasswordAuthenticationFilter(ObjectMapper objectMapper) {
+        super(DEFAULT_LOGIN_PATH_REQUEST_MATCHER);
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request,
+                                                HttpServletResponse response) throws AuthenticationException, IOException {
+        if (request.getContentType() == null || !request.getContentType().equals(CONTENT_TYPE)) {
+            throw new AuthenticationServiceException("Authentication Content-Type not supported: " + request.getContentType());
+        }
+
+        String messageBody = StreamUtils.copyToString(request.getInputStream(), StandardCharsets.UTF_8);
+
+        Map<String, String> usernamePasswordMap = objectMapper.readValue(messageBody, Map.class);
+
+        String username = usernamePasswordMap.get(USERNAME_KEY);
+        String password = usernamePasswordMap.get(PASSWORD_KEY);
+
+        UsernamePasswordAuthenticationToken authRequest = new UsernamePasswordAuthenticationToken(username, password);
+
+        return this.getAuthenticationManager().authenticate(authRequest);
```

Comment:
p1. UsernamePasswordAuthenticationToken에 정적 팩토리 메소드 `unauthenticated(Object principal, Object credentials);`를 사용하면 좋을 것 같아요!

p2. 빈 줄 삭제

p2. 일회성 객체 바로 사용

저는 빈 줄이랑 일회성 객체를 변수로 생성하면 코드가 길어져서 가독성이 떨어지는 것 같아요. 아래와 같은 코드는 어떠신가요?

```suggestion
        Map<String, String> usernamePasswordMap = objectMapper.readValue(StreamUtils.copyToString(request.getInputStream(), StandardCharsets.UTF_8), Map.class);
        String username = usernamePasswordMap.get(USERNAME_KEY);
        String password = usernamePasswordMap.get(PASSWORD_KEY);
        return this.getAuthenticationManager().authenticate(UsernamePasswordAuthenticationToken.unauthenticated(username, password));
```